contracts/flow/staking/BloctoTokenStaking.cdc

```diff
@@ -445,6 +501,59 @@ pub contract BloctoTokenStaking {
         return BloctoTokenStaking.totalTokensStaked
     }
 
+    /// Epoch
+    pub fun getEpoch(): UInt64 {
+        let epoch = self.account.copy<UInt64>(from: /storage/bloctoTokenStakingEpoch)
+        if epoch == nil {
+            return 0
+        } else {
+            return epoch!
+        }
+    }
+
+    access(contract) fun setEpoch(_ epoch: UInt64) {
+        self.account.load<UInt64>(from: /storage/bloctoTokenStakingEpoch)
+        self.account.save<UInt64>(epoch, to: /storage/bloctoTokenStakingEpoch)
+    }
+
+
+    access(contract) fun getStakingRewardKey(epoch: UInt64, stakerID: UInt64): String {
+        // key: {EPOCH}_{STAKER_ID}
+        return epoch.toString().concat("_").concat(stakerID.toString())
+    }
+
+    /// staking reward records
+    pub fun hasSentStakingReward(epoch: UInt64, stakerID: UInt64): Bool {
+        let stakingRewardRecordsRef = self.account.copy<{String: Bool}>(from: /storage/bloctoTokenStakingStakingRewardRecords)
+        if stakingRewardRecordsRef == nil {
+            return false
+        }
+
+        let key = BloctoTokenStaking.getStakingRewardKey(epoch: epoch, stakerID: stakerID)
+        if stakingRewardRecordsRef![key] == nil {
+            return false
+        }
+
+        return stakingRewardRecordsRef![key]!
+    }
+
+    access(contract) fun insertStakingRewardRecord(epoch: UInt64, stakerID: UInt64) {
+        let key = BloctoTokenStaking.getStakingRewardKey(epoch: epoch, stakerID: stakerID)
+
+        let stakingRewardRecordsRef = self.account.load<{String: Bool}>(from: /storage/bloctoTokenStakingStakingRewardRecords)
```

Comment:
Is there an estimation how fast this grows?
in terms of storage size and FLOW/epoch

Reply:
0.0000099 FLOW for a user do his first stake.
0.00000067 FLOW per user for switching epoch